### ISSUE TITLE
react-unused-props-and-state doesn't understand usage in setState 

### ISSUE THREAD

**Issue by imsolost** [microsoft/tslint-microsoft-contrib#796]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 6.0.0
-   **TSLint version**: 5.12.1
-   **TypeScript version**: 3.2.4
-   **Running TSLint via**: CLI

#### TypeScript code being linted

https://github.com/imsolost/test-app/blob/master/src/App.tsx

```tsx
interface IProps {
  readonly firstProp: boolean
  readonly secondProp: boolean
  readonly thirdProp: boolean
}

// ...

class App extends React.Component<
  IProps,
  IState
> {

  // ...

  public async componentWillMount() {
    this.setState((_state, props) => ({
      message: 'test message',
      firstProp: props.firstProp,
      secondProp: props.secondProp,
      thirdProp: props.thirdProp,
    }))
  }
}
```

with `tslint.json` configuration: https://github.com/imsolost/test-app/blob/master/tslint.json

```json
{
  "rulesDirectory": [
    "node_modules/tslint-react/rules",
    "node_modules/tslint-microsoft-contrib/"
  ],
  "rules": {  
    "react-unused-props-and-state": [
      true,
      {
        "props-interface-regex": "Props$",
        "state-interface-regex": "State$"
      }
    ]
  }
}
```

#### Actual behavior
The props defined in the interface are flagged as unused by tslint with the following errors:

Unused React property defined in interface: firstProp (react-unused-props-and-state) 
Unused React property defined in interface: secondProp (react-unused-props-and-state) 
Unused React property defined in interface: thirdProp (react-unused-props-and-state) 

#### Expected behavior

These properties are used in the setState function so they should not trigger this error.

**Comment by JoshuaKGoldberg** [comment]
Great catch! The `(_state, props)` updater passed to `setState` is a valid way to access props, but the rule doesn't seem to know that. Accepting PRs to fix that bug.

**Comment by shiftkey** [comment]
I've got my eye on addressing this one after #824 lands, now that I know more about how the rule works.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #876. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!
